# ChemSpider

Experimental logP records of compounds in fragment-like set was checked by manually querying Chemspider website. Experimental logP values (if exists), were under `Properties/Experimental Physico-chemical Properties/Experimental LogP`. I recorded these values under Chemspider columb in `df_frag_final_chemspider_logP.csv`.

In [3]:
import pandas as pd
import pickle

In [4]:
df_chemspider = pd.read_csv("df_frag_final_chemspider_logP.csv")
df_chemspider

Unnamed: 0  eMolecules ID                     canonical isomeric SMILES  \
0            0        6679830                 c1cc2c(cc1O)c3c(o2)C(=O)NCCC3   
1            1         719540                  c1ccc(cc1)n2c3c(cn2)c(ncn3)N   
2            2       45490344                      c1ccc(cc1)C(=O)Nc2ccncc2   
3            3       27013168              CSc1ccc(cc1)/C=C/2\C(=O)NC(=O)N2   
4            4       37095168                    c1ccc2c(c1)ncn2c3ccc(cc3)O   
5            5       37053191            c1ccc(cc1)c2[nH]c3ccc(cc3n2)C(=O)N   
6            6       31653344                    c1ccc(cc1)n2cnc3c2ccc(c3)N   
7            7        8332960             c1ccc2c(c1)nc(n2CC3CC3)c4c(non4)N   
8            8       13419113              c1ccc(cc1)NC(=O)c2c3ccccc3[nH]n2   
9            9        1503405                    Cc1nnc2n1nc(s2)c3ccc(cc3)N   
10          10       23354217          c1ccc(cc1)C(=O)NCC(=O)Nc2nc3ccccc3s2   
11          11       17004732               c1cc(c(c(c1)Cl)C(=O)Nc2ccncc2)F   
12          12       18893169            c1cc2cccnc2c(c1)NC(=O)c3cc(cnc3)Br   
13          13        2266357              c1cc(c(c(c1)Cl)Cl)C(=O)Nc2ccncc2   
14          14        1865544                COc1cccc(c1)Nc2c3ccccc3ncn2.Cl   
15          15        1202048                       c1ccc(cc1)n2cnc3c2cccc3   
16          16        1444229                 c1ccc(cc1)CSc2nnc(o2)c3ccncc3   
17          17        3374919          c1ccc(cc1)Oc2ccc(cc2)C(=O)NCc3ccncc3   
18          18        1228629             c1ccc(cc1)Cc2nnc(s2)NC(=O)c3cccs3   
19          19       30719859              c1ccc2c(c1)c(ncn2)NCc3ccc(cc3)Cl   
20          20       18908671         c1ccc(c(c1)NC(=O)c2ccc(o2)Cl)N3CCCCC3   
21          21        1377874       Cc1ccc(cc1)S(=O)(=O)Nc2cccc(c2)C(F)(F)F   
22          22       25775231  c1cc(c(cc1c2ccc(o2)/C=C/3\C(=O)NC(=O)S3)Cl)F   
23          23        5784088       c1cc(ccc1c2nnc3n2nc(cc3)NC4CC4)C(F)(F)F   
24          24       45809595           CC(C)c1ccc(cc1)/C=C\2/c3ccccc3NC2=O   

                               eMolecules SMILES   pKas in [3,11]  XlogP  \
0                      Oc1cc2c3CCCNC(=O)c3oc2cc1          [9.119]  0.720   
1                         Nc1ncnc2c1cnn2c1ccccc1          [3.869]  1.499   
2                         O=C(c1ccccc1)Nc1ccncc1          [4.714]  1.695   
3                 CSc1ccc(cc1)/C=C\1/NC(=O)NC1=O           [8.74]  1.857   
4                       Oc1ccc(cc1)n1cnc2c1cccc2    [5.82, 8.709]  2.219   
5             NC(=O)c1ccc2c(c1)nc([nH]2)c1ccccc1          [6.342]  2.192   
6                       Nc1ccc2c(c1)ncn2c1ccccc1          [6.348]  2.333   
7                  Nc1nonc1c1nc2c(n1CC1CC1)cccc2          [4.082]  2.440   
8                 O=C(c1n[nH]c2c1cccc2)Nc1ccccc1          [8.398]  2.643   
9                     Nc1ccc(cc1)c1nn2c(s1)nnc2C    [3.49, 5.213]  2.601   
10          O=C(Nc1nc2c(s1)cccc2)CNC(=O)c1ccccc1          [8.672]  2.718   
11                   O=C(c1c(F)cccc1Cl)Nc1ccncc1  [4.714, 10.192]  2.786   
12              Brc1cncc(c1)C(=O)Nc1cccc2c1nccc2  [4.001, 10.328]  2.923   
13                  O=C(c1cccc(c1Cl)Cl)Nc1ccncc1  [4.714, 10.167]  2.927   
14                COc1cccc(c1)Nc1ncnc2c1cccc2.Cl           [4.05]  3.139   
15                       c1ccc(cc1)n1cnc2c1cccc2           [5.74]  3.105   
16                 c1ccc(cc1)CSc1nnc(o1)c1ccncc1          [4.902]  3.306   
17            O=C(c1ccc(cc1)Oc1ccccc1)NCc1ccncc1          [5.124]  3.241   
18               O=C(c1cccs1)Nc1nnc(s1)Cc1ccccc1           [7.12]  3.605   
19                  Clc1ccc(cc1)CNc1ncnc2c1cccc2          [5.564]  3.523   
20             Clc1ccc(o1)C(=O)Nc1ccccc1N1CCCCC1          [5.346]  3.795   
21       Cc1ccc(cc1)S(=O)(=O)Nc1cccc(c1)C(F)(F)F          [7.903]  3.748   
22  O=C1S/C(=C/c2ccc(o2)c2ccc(c(c2)Cl)F)/C(=O)N1          [8.052]  3.908   
23       FC(c1ccc(cc1)c1nnc2n1nc(cc2)NC1CC1)(F)F          [4.394]  4.129   
24         O=C1Nc2c(/C/1=C/c1ccc(cc1)C(C)C)cccc2 

In [5]:
df_chemspider_exp_logP = df_chemspider.loc[df_chemspider["Chemspider"] != "None"]
print("{} chemicals have experimental logP reprted in ChemSpider.".format(df_chemspider_exp_logP.shape[0]))

7 chemicals have experimental logP reprted in ChemSpider.


I will save lists of chemicals selected for fragment-like set that have reported experimental logP in ChemSpider in lists (pickle files) for each round. eMolecules IDs below are listed here because they are detected to have experimental logP available.

#### Round 1 
[490595, 532754, 4934119, 2867544, 2727697, 493608, 508324, 2042282, 1698122, 703997]

#### Round 2
[45490344, 27013168, 1503405, 17004732, 2266357, 1202048, 3374919]

In [6]:
fragments_with_exp_logP = list(df_chemspider_exp_logP.loc[:,"eMolecules ID"])
filename = "fragments_with_exp_logP_round2.pickle"
pickle.dump(fragments_with_exp_logP, open(filename, "wb"))

In [7]:
fragments_with_exp_logP_list = pickle.load(open(filename, "rb"))
print(fragments_with_exp_logP_list)

[45490344, 27013168, 1503405, 17004732, 2266357, 1202048, 3374919]
